frontend/src/components/Pagination.tsx

```diff
@@ -70,11 +71,16 @@ const Pagination: React.FC<PaginationProps> = ({
         {pageNumbers.map((number, index) => (
           <React.Fragment key={index}>
             {number === '...' ? (
-              <span className="flex h-10 w-10 items-center justify-center text-gray-600 dark:text-gray-400">
+              <span className="flex h-10 w-10 items-center justify-center text-gray-600 dark:text-gray-400"
+              data-testid="ellipsis"
```

Comment:
Actual components should not have a `data-testid` set on the elements. Please check other test files to see how to address this.